src/librustc_trans/trans/controlflow.rs

```diff
@@ -171,16 +170,7 @@ pub fn trans_if<'blk, 'tcx>(bcx: Block<'blk, 'tcx>,
             // if true { .. } [else { .. }]
             bcx = trans_block(bcx, &*thn, dest);
             trans::debuginfo::clear_source_location(bcx.fcx);
-
-            if let Some(elexpr) = els {
-                let mut trans = TransItemVisitor { ccx: bcx.fcx.ccx };
-                trans.visit_expr(&*elexpr);
-            }
```

Comment:
I vaguely remember that a long time ago this was required for something like:

```rust
// foo.rs
pub fn foo<T>() {
    if true { 
    } else {
        println!("test");
    }
}

// bar.rs
extern crate foo;
fn main() {
    foo::foo<int>();
}
```

The monomorphization of `foo` in `bar.rs` caused something to reference the statics produced by `println!`, assuming they were available in `foo.o` (but we didn't translate them due to the constant branch).

I may also be remembering something totally bogus though! I suspect if this passes all tests it's fine anyway.